bgpd/bgp_attr.c

```diff
@@ -3015,11 +3022,12 @@ size_t bgp_packet_mpattr_start(struct stream *s, struct peer *peer, afi_t afi,
 					   IPV6_MAX_BYTELEN);
 				stream_put(s, &attr->mp_nexthop_local,
 					   IPV6_MAX_BYTELEN);
-			} else {
+			} else if (attr->mp_nexthop_len != 0) {
 				stream_putc(s, IPV6_MAX_BYTELEN);
 				stream_put(s, &attr->mp_nexthop_global,
 					   IPV6_MAX_BYTELEN);
-			}
+			} else if (safi == SAFI_FLOWSPEC)
+				stream_putc(s, 0); /* no nexthop for flowspec */
```

Comment:
so flowspec only exists in v6 under evpn? just making certain I'm reading this correctly...

Reply:
this makes sense -- it's just preventing it from being pulled for SR.

Reply:
+1, dead code needs to be removed; safi == SAFI_FLOWSPEC always false